src/_guides/language/effective-dart/design.md

```diff
@@ -1003,24 +1035,27 @@ The language specifies that this check is done automatically and your `==`
 method is called only if the right-hand side is not `null`.
 
 {:.good-style}
+<?code-excerpt "misc/lib/effective_dart/design_good.dart (eq-dont-check-for-null)" replace="/operator ==|hashCode/[!$&!]/g"?>
 {% prettify dart %}
 class Person {
   final String name;
+  // ...
+  @override
+  [!operator ==!](other) => other is Person && name == other.name;
 
-  operator ==(other) =>
-      other is Person && name == other.name;
+  @override
+  int get [!hashCode!] => ...
 }
 {% endprettify %}
 
 {:.bad-style}
+<?code-excerpt "misc/lib/effective_dart/design_bad.dart (eq-dont-check-for-null)" replace="/\w+ != null/[!$&!]/g"?>
 {% prettify dart %}
 class Person {
   final String name;
-
-  operator ==(other) =>
-      other != null &&
-      other is Person &&
-      name == other.name;
+  // ···
+  @override
+  operator ==(other) => [!other != null!] && other is Person && ...
```

Comment:
I'm unsure that you'll like the highlight here, but it puts emphasis on the bad part.

Reply:
This is actually a somewhat bad example because `other != null && other is Person` is already redundant because `null is! Person`. It's bad code, independently of it being in a `==` declaration.

The bad thing here is to compare to `null` at all in `==` because we know the argument will never be `null`. If possible, we should focus on that.

I don't have a good example because I very rarely make a `==` method that doesn't start with an `is` check.

Reply:
> The bad thing here is to compare to `null` at all in `==` because we know the argument will never be `null`. If possible, we should focus on that.

To focus on that I suggest that we elide the rest of the expression, like this:

```dart
operator ==(other) => other != null && ...
```

Reply:
LGTM